internal/wasm/sdk/internal/wasm/vm.go

```diff
@@ -238,9 +252,88 @@ func newVM(opts vmOpts) (*VM, error) {
 	return v, nil
 }
 
+func getABIVersion(i *wasmtime.Instance, store wasmtime.Storelike) (int32, int32, error) {
+	major := i.GetExport(store, "opa_wasm_abi_version").Global()
+	minor := i.GetExport(store, "opa_wasm_abi_minor_version").Global()
+	if major != nil && minor != nil {
+		majorVal := major.Get(store)
+		minorVal := minor.Get(store)
+		if majorVal.Kind() == wasmtime.KindI32 && minorVal.Kind() == wasmtime.KindI32 {
+			return majorVal.I32(), minorVal.I32(), nil
+		}
+	}
+	return 0, 0, fmt.Errorf("couldn't retrieve ABI version")
```

Comment:
nit: I'm unsure what error actually gets printed but this would seem to indicate an invalid wasm module got passed into the VM... I'm wondering if something like `invalid module: failed to read ABI version" would be more helpful.